### ISSUE TITLE
Routes ending with -ves search for models ending in -fe

### ISSUE THREAD

**Issue by nihalgonsalves** [JSONAPI-Resources/jsonapi-resources#885]
Steps to reproduce:
1. Clone `https://github.com/cerebris/peeps`, bundle, rake, migrate
2. Add a line such as `jsonapi_resources :drives` to `config/routes.rb` (notice the ending, driVES)
3. `rails s`

(I did this on an already existing project with the correct resources and controller files. But the peeps project is enough to demonstrate the search error)

Error:

```
jsonapi-resources-0.7.0/lib/jsonapi/resource.rb:310:in `resource_for':
JSONAPI: Could not find resource 'drives'.
(Class DrifeResource not found) (NameError)
```

(Notice that for the resource 'drives' jsonapi-resources is looking for 'DrifeResource')

**Comment by lgebhardt** [comment]
JR uses `ActiveSupport::Inflector`. I'm guessing this is related to the inflection rules for your system. Could you try the test of 

``` ruby
"drive".pluralize
```

from somewhere in your project. My guess is it will pluralize to drife.

**Comment by mroach** [comment]
@lgebhardt is right on there. It's bizarre that it convert drives to drife. Add this to `config/initializers/inflections.rb` to correct the inflection of "drives" and anything else you need to fix.

``` ruby
ActiveSupport::Inflector.inflections(:en) do |inflect|
  inflect.singular /^(drive)s$/i, '\1'
end
```

**Comment by nihalgonsalves** [comment]
Yep, that seems to be the problem. The thing is, there isn't anything specific to my project that would cause such an error. (`inflections.rb` is empty, too). Looks like this could be a `ActiveSupport::Inflector` bug then? Bizarre indeed!

**Comment by mroach** [comment]
I was just browsing the source for the ActiveSupport inflectors and the comment at the heading of it says the list is not complete and it is frozen. It's intended as a starting point for applications and the developer should override as necessary. They won't accept any PRs or fixes because it could break existing applications that rely on incorrect inflections. Although that's what versioning and tests are for... 🙄

**Comment by nihalgonsalves** [comment]
Yes, I just came across the same thing - crazy. I'm new to Rails and this really confused me for a bit. I figured that it was some pluralisation rule that was causing the problem, but for a word like 'drive', it is quite surprising! I opened this issue here because I didn't think that a project the size of Rails could possibly have such a bug for a normal word, but I guess I was wrong..

Thanks 👍
